File: src/test/java/com/baulsupp/oksocial/apidocs/ServiceApiDocPresenterTest.java
package com.baulsupp.oksocial.apidocs;

import com.baulsupp.oksocial.authenticator.ServiceInterceptor;
import com.baulsupp.oksocial.credentials.CredentialsStore;
import ee.schimke.oksocial.output.TestOutputHandler;
import java.io.IOException;
import okhttp3.OkHttpClient;
import org.junit.jupiter.api.Test;

import static com.google.common.collect.Lists.newArrayList;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class ServiceApiDocPresenterTest {
  private TestOutputHandler outputHandler = new TestOutputHandler();
  private OkHttpClient client = new OkHttpClient();
  private CredentialsStore credentialsStore = CredentialsStore.NONE;
  private ServiceApiDocPresenter presenter =
      new ServiceApiDocPresenter(new ServiceInterceptor(client, credentialsStore), client,
          credentialsStore);

  @Test public void returnsAllUrls() throws IOException {
    presenter.explainApi("https://api1.test.com/me", outputHandler, client);

    assertEquals(newArrayList("Test: https://api1.test.com/me"), outputHandler.stdout);
  }

  @Test public void errorForUnknown() throws IOException {
    presenter.explainApi("https://api1.blah.com/me", outputHandler, client);

    assertEquals(newArrayList("No documentation for: https://api1.blah.com/me"),
        outputHandler.stdout);
  }
}

File: src/test/java/com/baulsupp/oksocial/apidocs/ServiceApiDocPresenterTest.kt
package com.baulsupp.oksocial.apidocs

import com.baulsupp.oksocial.authenticator.ServiceInterceptor
import com.baulsupp.oksocial.credentials.CredentialsStore
import com.baulsupp.oksocial.output.TestOutputHandler
import com.google.common.collect.Lists.newArrayList
import okhttp3.OkHttpClient
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.io.IOException

class ServiceApiDocPresenterTest {
  private val outputHandler = TestOutputHandler<Any>()
  private val client = OkHttpClient()
  private val credentialsStore = CredentialsStore.NONE
  private val presenter = ServiceApiDocPresenter(ServiceInterceptor(client, credentialsStore),
      client,
      credentialsStore)

  @Test
  @Throws(IOException::class)
  fun returnsAllUrls() {
    presenter.explainApi("https://api1.test.com/me", outputHandler, client)

    assertEquals(newArrayList("Test: https://api1.test.com/me"), outputHandler.stdout)
  }

  @Test
  @Throws(IOException::class)
  fun errorForUnknown() {
    presenter.explainApi("https://api1.blah.com/me", outputHandler, client)

    assertEquals(newArrayList("No documentation for: https://api1.blah.com/me"),
        outputHandler.stdout)
  }
}
